src/parse/mod.rs

```diff
@@ -184,6 +184,16 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
                 "fatal: maximum recursion depth exceeded ({} > {})",
                 depth, MAX_DEPTH
             );
+            if !self.error_handler.is_empty() {
+                println!("pending errors:");
+                // we're about to crash, no point worrying about performance
+                for error in self.error_handler.clone() {
```

Comment:
Why not simple avoid this clone?

Reply:
`recursion_check` takes `&self`, not `&mut self`.

Reply:
But why not clone to a temp variable and use that error handler twice?